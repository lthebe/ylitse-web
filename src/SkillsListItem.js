import React from 'react';
import PropTypes from 'prop-types';
import Card, { CardContent } from 'material-ui/Card';
import IconButton from 'material-ui/IconButton';
import Typography from 'material-ui/Typography';
import DeleteIcon from 'material-ui-icons/Delete';
import { withStyles } from 'material-ui/styles';

const styles = theme => ({
    text: {
        noWrap: true,
        fontSize: 16,
    },
    card: {
        background: theme.palette.background.default,
        margin: [[5, 7, 5, 3]],
        paddingLeft: 8,
    },
    cardContent: {
        paddingTop: 6,
        paddingBottom: [6, '!important'],
    },
    iconButton: {
        '&:hover': {
            color: theme.palette.secondary.main,
        },
        marginLeft: 8,
        marginRight: -10,
    },
    icon: {
        fontSize: 22,
    },
});

const SkillsListItem = ({ label, classes, onDelete }) => (
    <Card
        className={classes.card}
    >
        <CardContent className={classes.cardContent}>
            <Typography
                className={classes.text}
            >
                {label}
                <IconButton
                    className={classes.iconButton}
                    onClick={onDelete}
                >
                    <DeleteIcon className={classes.icon} />
                </IconButton>
            </Typography>
        </CardContent>
    </Card>
);

SkillsListItem.propTypes = {
    label: PropTypes.string.isRequired,
    classes: PropTypes.shape({
        text: PropTypes.string,
        card: PropTypes.string,
        cardContent: PropTypes.string,
        iconButton: PropTypes.string,
        icon: PropTypes.string,
    }).isRequired,
    onDelete: PropTypes.func.isRequired,
};

export default withStyles(styles)(SkillsListItem);
