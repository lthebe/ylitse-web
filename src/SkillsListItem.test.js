import React from 'react';
import { shallow } from 'enzyme';

import SkillsListItem from './SkillsListItem';

describe('SkillListItem', () => {
    test('renders correctly', () => {
        const props = {
            label: 'MySkill', onDelete: () => {}, classes: {},
        };
        const wrapper = shallow(<SkillsListItem {...props} />).dive();

        expect(wrapper).toMatchSnapshot();
    });
});
