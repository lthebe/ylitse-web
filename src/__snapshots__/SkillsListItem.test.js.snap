// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkillListItem renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SkillsListItem
    classes={
      Object {
        "card": "SkillsListItem-card-2",
        "cardContent": "SkillsListItem-cardContent-3",
        "icon": "SkillsListItem-icon-5",
        "iconButton": "SkillsListItem-iconButton-4",
        "text": "SkillsListItem-text-1",
      }
    }
    label="MySkill"
    onDelete={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <WithStyles(CardContent)
        className="SkillsListItem-cardContent-3"
      >
        <WithStyles(Typography)
          className="SkillsListItem-text-1"
        >
          MySkill
          <WithStyles(IconButton)
            className="SkillsListItem-iconButton-4"
            onClick={[Function]}
          >
            <pure(Delete)
              className="SkillsListItem-icon-5"
            />
          </WithStyles(IconButton)>
        </WithStyles(Typography)>
      </WithStyles(CardContent)>,
      "className": "SkillsListItem-card-2",
      "raised": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Typography)
          className="SkillsListItem-text-1"
        >
          MySkill
          <WithStyles(IconButton)
            className="SkillsListItem-iconButton-4"
            onClick={[Function]}
          >
            <pure(Delete)
              className="SkillsListItem-icon-5"
            />
          </WithStyles(IconButton)>
        </WithStyles(Typography)>,
        "className": "SkillsListItem-cardContent-3",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "MySkill",
            <WithStyles(IconButton)
              className="SkillsListItem-iconButton-4"
              onClick={[Function]}
            >
              <pure(Delete)
                className="SkillsListItem-icon-5"
              />
            </WithStyles(IconButton)>,
          ],
          "className": "SkillsListItem-text-1",
        },
        "ref": null,
        "rendered": Array [
          "MySkill",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <pure(Delete)
                className="SkillsListItem-icon-5"
              />,
              "className": "SkillsListItem-iconButton-4",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "SkillsListItem-icon-5",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <WithStyles(CardContent)
          className="SkillsListItem-cardContent-3"
        >
          <WithStyles(Typography)
            className="SkillsListItem-text-1"
          >
            MySkill
            <WithStyles(IconButton)
              className="SkillsListItem-iconButton-4"
              onClick={[Function]}
            >
              <pure(Delete)
                className="SkillsListItem-icon-5"
              />
            </WithStyles(IconButton)>
          </WithStyles(Typography)>
        </WithStyles(CardContent)>,
        "className": "SkillsListItem-card-2",
        "raised": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Typography)
            className="SkillsListItem-text-1"
          >
            MySkill
            <WithStyles(IconButton)
              className="SkillsListItem-iconButton-4"
              onClick={[Function]}
            >
              <pure(Delete)
                className="SkillsListItem-icon-5"
              />
            </WithStyles(IconButton)>
          </WithStyles(Typography)>,
          "className": "SkillsListItem-cardContent-3",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "MySkill",
              <WithStyles(IconButton)
                className="SkillsListItem-iconButton-4"
                onClick={[Function]}
              >
                <pure(Delete)
                  className="SkillsListItem-icon-5"
                />
              </WithStyles(IconButton)>,
            ],
            "className": "SkillsListItem-text-1",
          },
          "ref": null,
          "rendered": Array [
            "MySkill",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <pure(Delete)
                  className="SkillsListItem-icon-5"
                />,
                "className": "SkillsListItem-iconButton-4",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "SkillsListItem-icon-5",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
